typescript/cli/src/deploy/utils.ts

```diff
@@ -186,3 +190,26 @@ function transformChainMetadataForDisplay(chainMetadata: ChainMetadata) {
     'Native Token: Decimals': chainMetadata.nativeToken?.decimals,
   };
 }
+
+/**
+ * Checks if the given chain technical stack is compatible with the core configuration.
+ *
+ * @param {ChainTechnicalStack | undefined} params.chainTechnicalStack - The technical stack of the chain.
+ * @param {CoreConfig} params.config - The core configuration to check.
+ * @returns {boolean} True if the configuration is compatible, false otherwise.
+ */
+export function isIsmCompatible({
```

Comment:
This function only checks the top-level config, but I think it should also make sure that there is no aggregation or static isms anywhere nested the config

Reply:
We added a check in HyperlaneIsmFactory for tech stack and Ism type

https://github.com/txfusion/hyperlane-monorepo/blob/3755332f6451b731618eac05e8200eb0ebbd9a5b/typescript/sdk/src/ism/HyperlaneIsmFactory.ts#L154